pkg/compose/hash.go

```diff
@@ -37,5 +37,12 @@ func ServiceHash(o types.ServiceConfig) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
+	for _, serviceConfig := range o.Configs {
+		if projectConfig, ok := project.Configs[serviceConfig.Source]; ok {
+			bytes = append(bytes, []byte(projectConfig.Content)...)
```

Comment:
This won't detect changes for config set by environment of file

Reply:
added support for env and config file